exonum/src/encoding/protobuf/mod.rs

```diff
@@ -53,17 +55,17 @@ pub trait ProtobufConvert: Sized {
 
 impl<T> BinaryForm for T
 where
-    T: ProtobufConvert,
-    <T as ProtobufConvert>::ProtoStruct: Message,
+    T: Message,
 {
     fn encode(&self) -> Result<Vec<u8>, Error> {
-        Ok(self.to_pb().write_to_bytes().unwrap())
+        Ok(self.write_to_bytes().unwrap())
```

Comment:
Is `self.write_to_bytes()` never returns error?

Reply:
It can but possible reasons are something like "unable to allocate memory" or smth. I thought that error handling will be revisited when someone will decide to remove old serialization, then all this method will be changed since encoding::Error probably will be gone.